sql/sql_acl.cc

```diff
@@ -14312,6 +14313,50 @@ bool acl_authenticate(THD *thd, uint com_change_user_pkt_len)
 
   if (res == CR_OK_HANDSHAKE_COMPLETE)
     thd->get_stmt_da()->disable_status();
+  else
+  if (redirect_enabled)
+  {
+    int total_size = 18
+                   + strlen(redirect_server_host)
+                   + 1 + strlen(redirect_server_port)
+                   + 6 + strlen(sctx->user)
+                   + (!strcmp(redirect_server_ttl, "0") ? 0 : (5 + strlen(redirect_server_ttl)));;
+
+    if (total_size >= MAX_REDIRECTION_LEN)
+    {
+      sql_print_error("error", ("redirection info is too large to return to client (len= %lu)",
+                     total_size));
+      my_ok(thd);
```

Comment:
is an ok the right response here?

Reply:
Tend to believe it is. Server needs the OK packet to tell client its request is authenticated and OK to proceed.

Reply:
the datatypes are wrong in more than a single case
strlen() gives size_t , but total_size is int, and in sql_print_error it is printed with %lu, which is for unsigned long.

sql_print_error() is wrong, as well. it is meant to be used like printf() - same parameters, not like DBUG_PRINT

Reply:
corrected miss in sql_print_error(); removed the long and dull size manipulation and turned to snprintf for a clearer view of what is written and whether the size exceeds limit.